flink-formats/flink-avro-glue-schema-registry/src/main/java/org/apache/flink/formats/avro/glue/schema/registry/GlueSchemaRegistryInputStreamDeserializer.java

```diff
@@ -73,12 +72,10 @@ public Schema getSchemaAndDeserializedStream(InputStream in) throws IOException
         in.read(inputBytes);
         in.reset();
 
-        MutableByteArrayInputStream mutableByteArrayInputStream = (MutableByteArrayInputStream) in;
         String schemaDefinition =
                 glueSchemaRegistryDeserializationFacade.getSchemaDefinition(inputBytes);
         byte[] deserializedBytes =
                 glueSchemaRegistryDeserializationFacade.getActualData(inputBytes);
-        mutableByteArrayInputStream.setBuffer(deserializedBytes);
```

Comment:
What was this actually doing? AFAICT, nothing?

Reply:
Yes, as far as I can tell as well... it is not doing anything, since the mutableByteArrayInputStream is not read further down the callchain... Compared with the implementations in other places for `flink-avro` as well. The buffer doesn't seem to be reset.